### ISSUE TITLE
Getting error while executing the playbook 

### ISSUE THREAD

**Issue by jyotibhanot** [pimvh/ssh_keygen#2]
I tried executing the playbook following the instructions on the README.md but getting below error 

```
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
ERROR! 'argument_specs' is not a valid attribute for a RoleMetadata

The error appears to be in '/root/ssh_keygen/meta/main.yaml': line 2, column 1, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

---
galaxy_info:
^ here
```

How can I resolve this?

**Comment by jyotibhanot** [comment]
This issue is resolved when I move the argument_specs to argument_specs.yaml

Also I have kept the password as blank still when it executes the role it ask for the password:

Enter passphrase (empty for no passphrase):
Enter same passphrase again:

**Comment by pimvh** [comment]
> This issue is resolved when I move the argument_specs to argument_specs.yaml
> 
> Also I have kept the password as blank still when it executes the role it ask for the password:
> 
> Enter passphrase (empty for no passphrase): Enter same passphrase again:

Hello, 

Thanks for opening an issue. To reply on first remark, I think you are getting an issue because of an running an older Ansible version:

```
Ansible also supports role specifications defined in the role meta/main.yml file, as well. However, any role that defines the specs within this file will not work on versions below 2.11. For this reason, we recommend using the meta/argument_specs.yml file to maintain backward compatibility.
```

I chose to directly adopt the modern version, because I am running everything on a very recent version of Ansible (2.14, also see `meta/main.yaml` with the minimal version, apparently that is not enforced), and my roles might not be compatible due to other reasons anyway. See [the official Ansible documentation](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_reuse_roles.html#role-argument-validation) for more context.

To reply on you second remark, It is intended behaviour that when you do not pass a password, it prompts. Would you like to generate SSH keys without passwords? Because the current role doesn't support that.

**Comment by jyotibhanot** [comment]
Thanks

**Comment by pimvh** [comment]
I am not sure, but you might have installed `ansible` via `apt` (if you are on debian-based distro like `Ubuntu`). I think the recommendation is using `pip` to install it, as that installs a more recent version (`apt` packages are generally lower versions due to the way debian values stability and code auditing see: [Debian docs](https://www.debian.org/doc/manuals/debian-faq/choosing.en.html#s3.1.3)).

See: [Installing Ansible](https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html)

**Comment by jyotibhanot** [comment]
I have installed ansible using yum.

**Comment by jyotibhanot** [comment]
I am trying to extend your role by adding a task:

```- name: Convert the SSH pubkey to PKCS8
  ansible.builtin.command:
    cmd: >-
      ssh-keygen
       -e -m PKCS8
       -f "{{ ssh_keygen_keyname }}.pub" > "{{ ssh_keygen_pkcs_key }}.pub"      
    chdir: "{{ ssh_keygen_storagedir }}"
    creates: "{{ ssh_keygen_storagedir }}/{{ ssh_keygen_pkcs_key }}.pub"
  changed_when: true
  when: not _ssh_key_path.stat.exists```

It does not work by giving error 'too many arguments`
Tryin to automate using ansible:

```# Generate the keypair
PRIV_KEY="sa-signer.key"
PUB_KEY="sa-signer.key.pub"
PKCS_KEY="sa-signer-pkcs8.pub"
# Generate a key pair
ssh-keygen -t rsa -b 2048 -f $PRIV_KEY -m pem
# convert the SSH pubkey to PKCS8
ssh-keygen -e -m PKCS8 -f $PUB_KEY > $PKCS_KEY```

I know it is out of the box but could you point out what I am doing wrong?

**Comment by pimvh** [comment]
Hello, I feel that this is moving away from an issue and turning more into a discussion. I think the original issue which you opened has be resolved. Could you close this issue, if you agree?

I would prefer to move to a discussion instead if that is possible. Feel free to open one.

**Comment by pimvh** [comment]
> This issue is resolved when I move the argument_specs to argument_specs.yaml
> 
> Also I have kept the password as blank still when it executes the role it ask for the password:
> 
> Enter passphrase (empty for no passphrase): Enter same passphrase again:

I just merged support for empty passphrases in #3
